raiden/network/rpc/client.py

```diff
@@ -474,7 +495,14 @@ def approve(self, contract_address, allowance):
             startgas=self.startgas,
             gasprice=self.gasprice,
         )
-        self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
+
+        try:
+            self.client.poll(transaction_hash.decode('hex'), timeout=self.poll_timeout)
+        except JSONRPCPollTimeoutException:
+            raise BlockchainPollTimeout()
+        except InvalidTransaction:
+            raise BlockchainInvalidTransaction()
```

Comment:
I see where you are coming from @ezdac but I don't believe there is any merit in duplicating exception code and namespacing them again inside Raiden. It's better to indeed re-raise the already formulated ethereum exeptions and thus have the actual information of the error easily propagated up in the call stack